tap_google_ads/streams.py

```diff
@@ -481,13 +488,14 @@ def sync(self, sdk_client, customer, stream, config, state):
 
                     singer.write_record(stream_name, record)
 
-            singer.write_bookmark(state, stream_name, replication_key, utils.strftime(query_date))
+            new_bookmark_value = {replication_key: utils.strftime(query_date)}
+            singer.write_bookmark(state, stream["tap_stream_id"], customer["customerId"], new_bookmark_value)
 
             singer.write_state(state)
 
             query_date += timedelta(days=1)
 
-        state = singer.bookmarks.set_currently_syncing(state, None)
+        state = singer.bookmarks.set_currently_syncing(state, (None, None))
```

Comment:
If I'm reading this right, it's meant to null out the currently syncing after each report finishes? Traditionally we've only nulled out the value once the whole sync finished so that if it gets interrupted in between streams it won't start over at the beginning. Looks like this might be running that risk.

Reply:
Fixing